share/functions/__fish_describe_command.fish

```diff
@@ -6,15 +6,25 @@
 # The whatis database is non-existent, so apropos tries (and fails) to create it every time,
 # which takes about half a second.
 #
-# So we disable this entirely in that case.
-if test (uname) = Darwin
-    set -l darwin_version (uname -r | string split .)
-    # macOS 15 is Darwin 19, this is an issue up to and including 10.15.3.
-    if test "$darwin_version[1]" = 19 -a "$darwin_version[2]" -le 3
-        function __fish_describe_command
+# Instead, we build a whatis database in the user home directory
+# and use grep to query it
+#
+# A function `__fish_apropos_update` is provided for updating
+#
+function __fish_apropos -d "Internal command to query the apopos database"
+    if test (uname) = Darwin
+        set db ~/.whatis.db
+
+        function __fish_apropos_update -d "Updates the apropos database $db on Macos"
+            echo "updating apropos / whatis database at $db"
+            man --path | tr ":" " " | xargs /usr/libexec/makewhatis -o $db
         end
-        # (remember: exit when `source`ing only exits the file, not the shell)
-        exit
+
+        [ -f $db ] || __fish_apropos_update
```

Comment:
There are two hard things in computer science:

- cache invalidation
- naming things
- off-by-one errors

This falls afoul of number 1 (or is that 0?). This will never ever update the cache, as long as fish is installed and the user doesn't delete the file. And we don't want to force the user to remember to refresh that file, that's not how we roll.

Ideally (well, ideally ideally apple would fix their stuff, but...) this could check if the man path has changed (new directories or new or changed files) - since macOS ships GNU test IIRC, it might be possible with `command test -ot` - that compares mtimes between files.

Unfortunately I don't think directory mtime works here (is that even a thing? I'm not sure), so you might have to check all files, and that might take too long.

The alternative is to do what some of our package completions do and only regenerate the file once a day or so. See e.g. https://github.com/fish-shell/fish-shell/blob/a1fd9e1b851bd876fb70da2743623f015d955935/share/functions/__fish_print_packages.fish#L63-L81

Reply:
The once a day solution seems the way to go. Will add this.